deps/azure-core-c/inc/azure/core/az_precondition.h

```diff
@@ -45,7 +45,7 @@
  * @brief Defines the signature of the callback function that application developers can write in
  * order to override the default precondition failure behavior.
  */
-typedef void (*az_precondition_failed_fn)();
+typedef void (*az_precondition_failed_fn)(void);
```

Comment:
>void [](start = 42, length = 4)

The warnings and errors settings from ulib a sdk are different, we have a task in backlog to unify it. As soon as we fix it, we will revisit this function prototype to make sure that it has the same format as in the SDK.